### ISSUE TITLE
writer.append() raises KeyError: '/Fields'

### ISSUE THREAD

**Issue by ZTThomasOberthanner** [py-pdf/pypdf#1627]
I tried to append a single PDF to a writer.

## Environment

```bash
$ python -m platform
Linux-6.1.9-1-MANJARO-x86_64-with-glibc2.36

$ python -c "import pypdf;print(pypdf.__version__)"
3.4.0
```

## Code + PDF

This is a minimal, complete example that shows the issue:

```python
writer = PdfWriter()
writer.append("test.pdf")
```

[test.pdf](https://github.com/py-pdf/pypdf/files/10715624/test.pdf)

## Traceback

```
Traceback (most recent call last):
  File "/home/username/PycharmProjects/m16db/play.py", line 10, in <module>
    writer.append("/home/username/PycharmProjects/m16db_testdata/test.pdf", excluded_fields=("/Annots",))
  File "/home/username/.local/share/virtualenvs/venv/lib/python3.10/site-packages/pypdf/_writer.py", line 2476, in append
    self.merge(
  File "/home/username/.local/share/virtualenvs/venv/lib/python3.10/site-packages/pypdf/_utils.py", line 441, in wrapper
    return func(*args, **kwargs)
  File "/home/username/.local/share/virtualenvs/venv/lib/python3.10/site-packages/pypdf/_writer.py", line 2623, in merge
    for f in reader.trailer["/Root"]["/AcroForm"]["/Fields"]:  # type: ignore
  File "/home/username/.local/share/virtualenvs/venv/lib/python3.10/site-packages/pypdf/generic/_data_structures.py", line 269, in __getitem__
    return dict.__getitem__(self, key).get_object()
KeyError: '/Fields'
```

**Comment by pubpub-zz** [comment]
The PDF is not in accordance with the standard:
![image](https://user-images.githubusercontent.com/4083478/218307947-aa42768d-aa92-45d3-8ca5-b725e738abfb.png)

Can you clarify which program you've used to generate it ?

**Comment by pubpub-zz** [comment]
As AcrobatReader is tolerant, I've preferred to add a robustness fix

**Comment by ZTThomasOberthanner** [comment]
Pdf was exported from Libre Office. Foxit Reader was used to add annotion.
Thank you very much for the quick fix!

**Comment by MartinThoma** [comment]
The fix was just merged. It will be part of `pypdf>3.4.1` which will be released latest 26.02.2023 (tomorrow)
